xwiki-commons-core/xwiki-commons-velocity/src/main/java/org/xwiki/velocity/internal/InternalVelocityEngine.java

```diff
@@ -37,34 +35,30 @@
 import org.apache.velocity.Template;
 import org.apache.velocity.VelocityContext;
 import org.apache.velocity.context.Context;
-import org.apache.velocity.runtime.RuntimeConstants;
 import org.apache.velocity.runtime.RuntimeInstance;
-import org.apache.velocity.runtime.directive.StopCommand;
 import org.apache.velocity.runtime.resource.Resource;
 import org.slf4j.Logger;
 import org.xwiki.component.annotation.Component;
 import org.xwiki.component.annotation.InstantiationStrategy;
 import org.xwiki.component.descriptor.ComponentInstantiationStrategy;
-import org.xwiki.component.manager.ComponentManager;
 import org.xwiki.context.Execution;
 import org.xwiki.context.ExecutionContext;
-import org.xwiki.velocity.VelocityConfiguration;
 import org.xwiki.velocity.VelocityContextFactory;
 import org.xwiki.velocity.VelocityEngine;
 import org.xwiki.velocity.VelocityTemplate;
 import org.xwiki.velocity.XWikiVelocityException;
-import org.xwiki.velocity.internal.directive.TryCatchDirective;
 
 /**
  * Default implementation of the Velocity service which initializes the Velocity system using configuration values
  * defined in the component's configuration. Note that the {@link #initialize} method has to be executed before any
  * other method can be called.
  *
  * @version $Id$
+ * @since 15.9-rc-1
  */
-@Component
+@Component(roles = InternalVelocityEngine.class)
```

Comment:
Using VelocityEngine as a component role is deprecated now and where InternalVelocityEngine is used I want to be sure it's a InternalVelocityEngine.

Reply:
> Using VelocityEngine as a component role is deprecated now

So shouldn't the interface itself be deprecated?

Reply:
At the very least it should be documented in the interface IMO. I feel like to be completely clean we'd need a new interface without the `@Role` and deprecate the current one, but maybe it's too much work for little benefits.

Reply:
> So shouldn't the interface itself be deprecated?

No it should not since that's what you are supposed to manipulate. You should just not directly try to inject one (it was already the case in practice before, you would not get a fully initialized engine).

> I feel like to be completely clean we'd need a new interface without the @Role and deprecate the current one, but maybe it's too much work for little benefits.

This interface never been used as component in practice (it never really provided a fully initialized engine), so I don't think it's needed.

I will just add a note that it should not be injected.

Reply:
ok

Reply:
Actually: why not officially breaking it and remove the `@Role` annotation then?

Reply:
Not a fan of any breakage I can technically avoid, that's all.

Reply:
in general me neither, but here it feels that it would be the cleanest option on the long run. And according to your comment it's probably harmless. To be checked with other committers maybe.